scripts/deno/ophan-components.ts

```diff
@@ -0,0 +1,165 @@
+import { Octokit } from "https://cdn.skypack.dev/octokit?dts";
+
+const attributes = ["data-link-name", "data-component"] as const;
+type OphanAttribute = typeof attributes[number];
```

Comment:
why is it `attributes[*number*]`, I'm finding this part quite hard to compute!

Reply:
Ah yes, this is one of the weirdness of Typescript, but is also very convenient for runtime type-checking.

The way you access an array is `array[0]`, and because we’ve declared that `attributes as const`, Typescript is able to determine that any call in form of `attribute[number]` will yield one of its members, it can create the union type `"data-link-name" | "data-component"`. The benefit of using this syntax is that checking if a `string` is an `OphanAttribute` is as simple as doing `attributes.includes(string)`, which is what the `isOphanAttribute` type predicate does. See [Typescript Playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAhgJwXAnjAvDA2gIjjgGhhwCNDjgcBdAbhgHp6ZoEBLMAcyyoFgAoUJFiJkKAIIQAwuGgZsecqUWUq8CDEHQ6jGAgCmcACbgANmlz4iSqyv78oKAA56YAGRAgILzA+cgAZvBIqFhgAK4AtiR6CLQMTCzsHPZOLgByIiAA7nK+egFBohLSQqGR0bHaTAowAD7EZHUUOHYCMrCsEBlI2XIAFBAAXMxQbJwAlMPqnTDdCL3oAHyFqMXtAHTswCZhhnoQA2qzmVnjNK0WigBe1OsRcI6HSxoyICZ66yYgHH0ABjgAEgA3hAAL44GDAzpzbIDcYwAD8xBmYBOMEAvBuAMT2IcMcDMvp4XOjAKJ7OFBv3G434OhqKLRWOp1RuMHxHi8GOJQA).

```ts
const array = ["a", "b", "c"]; // string[]
const arrayAsConst = ["a", "b", "c"] as const; // readonly ["a", "b", "c"]

type Loose = typeof array[number]; // string
type Narrow = typeof arrayAsConst[number]; // "a" | "b" | "c"

const isNarrow = (s: string): s is Narrow => arrayAsConst.includes(s as Narrow);

["a", "z"].map(s => console.log(`"${s}" ${isNarrow(s) ? "is narrow 🙆" : "is loose 🙅"}`))
// "a" is narrow 🙆
// "z" is loose 🙅
```

Reply:
😰😰😰 Definitely a confusing syntax! It feels like there should be a more readable (atleast for typescript non-experts like me), because I agree that having the ability to do `attributes.includes(string)` is great, but when I'd tried to work backwards to see _why_ you were able to do that, it made little sense to me!

Thanks for the great explanation (I'm still not quite sure I understand, but thats on me!) :)

Reply:
I have managed to get away from this syntax for the time being, so hopefully this is a problem for another day.

I am aware that this file could benefit from being a little more accessible by favouring simplicity. As this is not in the critical path of DCR rendering and are bound by their own set of TS rules, I feel tempted to indulge in slightly more advanced TS features. Happy to try an refactor for simplicity, but maybe in a subsequent PR?